tests/template/t13515.nim

```diff
@@ -0,0 +1,16 @@
+discard """
+  action: compile
+"""
+
+template test: bool = true
+
+# compiles:
+if not test:
+  echo "wtf"
+
+# does not compile:
+template x =
+  if not test:
```

Comment:
Why does it matter that `test`'s typ field is set back to nil? It's set to `bool` and that's ok, no need to re-sem it.